content/doc/tutorials/using-jenkinsfile-runner-github-action-to-build-jenkins-pipeline.adoc

```diff
@@ -0,0 +1,378 @@
+  jcasc: jcasc.yml
+----
+
+For additional information, refer to the link:https://github.com/jenkinsci/configuration-as-code-plugin/tree/master/demos[examples] provided by the configuration-as-code-plugin, and learn how to configure the Jenkins instance without using the UI page. 
+Some plugins do not have concrete examples, but you can debug and find their JCasC in the UI page. 
+You can check the configuration in *Manage Jenkins* -> *Configuration as Code* -> *View Configuration*. 
+Then, you can copy the parts you need to the JCasC file.
+
+=== Add and configure plugins (Optional)
+
+There are many powerful plugins that can be part of your Jenkins instance. 
+You can add the plugins in the plugin list file, and configure the plugins in the JCasC YAML file as needed.
+
+In this example, we review how to install JDK11 in the ephemeral Jenkins instance.
+
+. Specify `adoptopenjdk` plugin in the plugins.txt file. 
+As the version is not specified, the latest version will be installed.
++
+[source,txt]
+----
+adoptopenjdk
+----
+. Create a JCasC Yaml file called jcasc.yml and specify which JDK version to install.
++
+[source,yaml]
+----
+tool:
+  jdk:
+    installations:
+      - name: jdk11
+        home: "~/jdk11"
+        properties:
+          - installSource:
+              installers:
+                - adoptOpenJdkInstaller:
+                    id: "jdk-11.0.14.1+1"
+----
+. Create a Jenkinsfile. Remember to set up the JDK11 as a tool.
++
+[source,groovy]
+----
+pipeline {
+    agent any
+    tools {
+        maven 'maven'
+        jdk 'jdk11'
+    }
+    stages {
+        stage('env') {
+            steps {
+                sh 'mvn --version'
+            }
+        }
+        stage('build') {
+            steps {
+                sh 'mvn clean install -B --no-transfer-progress'
+            }
+        }
+    }
+}
+----
+. Specify the jcasc.yml in the GitHub Actions input.
++
+[source,yaml]
+----
+uses:
+  jenkinsci/jfr-container-action@master
+with:
+  command: run
+  jenkinsfile: Jenkinsfile
+  pluginstxt: plugins.txt
+  jcasc: jcasc.yml
+----
+
+=== Configure ephemeral Jenkins instance (Optional)
+
+Sometimes, JCasC might not be able to provide the configurations you need. 
+In this case, refer to link:https://www.jenkins.io/doc/book/managing/groovy-hook-scripts/[Groovy Hook Scripts] to set up the ephemeral Jenkins instance. 
+These Groovy scripts will have full access to the ephemeral Jenkins server and will be executed right after Jenkins starts up.
+
+NOTE: This option and its core are still in progress, so it’s not mentioned in the Jenkinsfile Runner GitHub Actions official guide.
+However, it does work and can be used at this time.
+
+In this example, we review how to use Groovy scripts to set up the Jenkins instance:
+
+. Create a directory, for example “groovy.init.d”, to store all your Groovy setup scripts.
+. Create a Groovy file called “test.groovy”.
+* Do not name it `init.groovy` because this name is already occupied.
```

Comment:
```suggestion
. Create a directory, for example `groovy.init.d`, to store all your Groovy setup scripts.
. Create a Groovy file called `test.groovy`.
* Do not name it `init.groovy` because this name is already occupied.
```
update for entry text